rocAL/rocAL/source/coco_meta_data_reader.cpp

```diff
@@ -83,105 +83,217 @@ void COCOMetaDataReader::print_map_contents()
+            }
+        }
+        else if (0 == std::strcmp(key, "annotations"))
+        {
+            RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+            parser.EnterArray();
+            while (parser.NextArrayValue())
+            {
+                int id = 1, label = 0;
+                std::array<float, 4> bbox;
+                if (parser.PeekType() != kObjectType)
+                {
+                    continue;
+                }
+                parser.EnterObject();
+                while (const char *internal_key = parser.NextObjectKey())
+                {
+                    if (0 == std::strcmp(internal_key, "image_id"))
+                    {
+                        id = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "category_id"))
+                    {
+                        label = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "bbox"))
+                    {
+                        RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+                        parser.EnterArray();
+                        int i = 0;
+                        while (parser.NextArrayValue())
+                        {
+                            bbox[i] = parser.GetDouble();
+                            ++i;
+                        }
+                    }
+                    else
+                    {
+                        parser.SkipValue();
+                    }
+                }
+                char buffer[13];
+                sprintf(buffer, "%012d", id);
+                string str(buffer);
+                std::string file_name = str + ".jpg";
+
+                auto it = _map_img_sizes.find(file_name);
+                ImgSizes image_size = it->second; //Normalizing the co-ordinates & convert to "ltrb" format
+                box.l = bbox[0] / image_size[0].w;
+                box.t = bbox[1] / image_size[0].h;
+                box.r = (bbox[0] + bbox[2]) / image_size[0].w;
+                box.b = (bbox[1] + bbox[3]) / image_size[0].h;
+
+                bb_coords.push_back(box);
+                bb_labels.push_back(label);
+                add(file_name, bb_coords, bb_labels, image_size);
+                bb_coords.clear();
+                bb_labels.clear();
+            }
+        }
+        else
+        {
+            parser.SkipValue();
+        }
     }
-    
-    for (auto iterator = annotation.begin(); iterator != annotation.end(); iterator++)
+    for (auto &elem : _map_content)
     {
-        float box_x = (*iterator)["bbox"][0].asFloat();
-        float box_y = (*iterator)["bbox"][1].asFloat();
-        float box_w = (*iterator)["bbox"][2].asFloat();
-        float box_h = (*iterator)["bbox"][3].asFloat();
-        int label = (*iterator)["category_id"].asInt();
-        int id = (*iterator)["image_id"].asInt();
-        char buffer[13];
-        sprintf(buffer, "%012d", id);
```

Comment:
wondering it this will work with datasets other than coco. It is assuming all filenames are 13 char long